webrtc/src/rtp_transceiver/rtp_sender/mod.rs

```diff
@@ -454,24 +542,72 @@ impl RTCRtpSender {
 
         self.replace_track(None).await?;
 
-        {
-            let stream_info = self.stream_info.lock().await;
+        let encodings = self.track_encodings.lock().await;
+        for te in encodings.iter() {
+            let stream_info = te.stream_info.lock().await;
             self.interceptor.unbind_local_stream(&stream_info).await;
+            te.srtp_stream.close().await?
         }
-
-        self.srtp_stream.close().await
+        Ok(())
     }
 
     /// read reads incoming RTCP for this RTPReceiver
     pub async fn read(&self, b: &mut [u8]) -> Result<(usize, Attributes)> {
-        self.internal.read(b).await
+        if let Some(encoding) = self.first_encoding().await? {
```

Comment:
Each SSRC has a separate RTCP feedback/resend/etc context. Since there is one SSRC per simulcast part, I think that means we also need a separate RTCP state for each.

Reply:
You are right in this instance, RTCP streams carry state